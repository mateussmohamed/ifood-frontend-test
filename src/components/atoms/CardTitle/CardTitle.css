.card-title {
  width: 100%;
  margin-top: 10px;
  color: #fff;
  font-size: 14px;
  font-weight: 400;
}
