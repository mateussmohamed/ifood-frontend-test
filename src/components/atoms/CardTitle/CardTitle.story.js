import { storiesOf } from '@storybook/react';
import React from 'react';
import CardTitle from './index';

const stories = storiesOf('CardTitle', module);

stories.add('with title prop', () => (
  <div style={{ backgroundColor: '#000' }}>
    <CardTitle title="Galactic vibes" />
  </div>
));
