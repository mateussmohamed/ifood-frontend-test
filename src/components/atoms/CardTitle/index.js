import React from 'react';
import PropTypes from 'prop-types';

import './CardTitle.css';

const CardTitle = ({ title }) => <span className="card-title">{title}</span>;

CardTitle.propTypes = {
  title: PropTypes.string.isRequired,
};

export default CardTitle;
