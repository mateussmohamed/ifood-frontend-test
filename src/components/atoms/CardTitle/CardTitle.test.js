import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import CardTitle from './index';

test('CardTitle match to snapshot', () => {
  const component = renderer.create(<CardTitle title="Galactic vibes" />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('CardTitle contains Galactic vibes when passed title prop', () => {
  const component = shallow(<CardTitle title="Galactic vibes" />);
  const rendered = <span className="card-title">Galactic vibes</span>;
  expect(component.contains(rendered)).toBeTruthy();
});
