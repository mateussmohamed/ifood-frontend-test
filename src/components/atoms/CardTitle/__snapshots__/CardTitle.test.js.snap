// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTitle match to snapshot 1`] = `
<span
  className="card-title"
>
  Galactic vibes
</span>
`;
